### ISSUE TITLE
Podcast section enrichment

### ISSUE THREAD

**Issue by jessleenyc** [forem/forem#271]
We have let the podcast section get into disrepair a bit over time. There's not _that much_ technical debt, per se, but it just is not a well-maintained part of the site conceptually. And with a lot more critical things, it keeps getting pushed back.

I think it could lend itself quite well to some community maintenance because there's still a lot of promise there.

## Basic things that need to be better kept up

- Observability over broken podcast links. We need to pull from feeds so authors understand their shows are being downloaded. It's not the kind of thing we can just cache on our own servers, images. But when things are broken, we're not really monitoring that with any confidence.
- Podcast player is a bit janky
- Podcast liquid tag, which could be one of the cooler things, is also janky.
- Commenting system on podcasts is fairly shoddy and incomplete
- Better ongoing dialog with the folks who do these podcasts, and ongoing outreach and info on how to get on dev.to (even though it's probably cool to just host whatever, because of the standard practices with podcast players, we prefer to get consent and have the relationship) 
- The whole feature just needs more tender loving care (TLC) but could be pretty awesome

## Ways it could be more awesome, provided TLC is there
- Users could be tagged in podcasts they've appeared in, and it could go on their profile
- Users could follow podcasts and have them in their feed
- More interaction on the shows via the comments between the hosts and listeners
- Suggested podcast episodes next to popular articles
- Let users create their own podcasts quickly and easily without the burden of finding their own audience (kind of like the rest of dev.to's offering)
- Once we have a native app, there could be pre-downloading of podcasts
- More creativity thrown at the problem than what I'm describing here

## Bottom line

This could be great but with all the other things going on, it's hard for us to justify us getting to this without losing focus on more critical elements of the site. Very willing to identify folks who want to lean in and take on this project.

**Comment by maestromac** [comment]
Not stale.

**Comment by vinf100** [comment]
So @jessleenyc, to have a more specific list of some things that need to be done:

### Podcast Player
- Not able to click-and-drag to a specific timestamp. Can only click. This makes it very frustrating to play from a specific timestamp
- You can't close the player. When the red X is pressed, it instead clicks on the timeline.
- When clicking beyond what has already been loaded, usually one of two things happen:
1) The player loads towards where you clicked, and the player moves towards where you clicked, but less than how much was loaded. You may have to click the same spot multiple times, until the player reaches it.
2) The player moves where you actually clicked, but it then freezes for about 5 seconds before playing.

    Sometimes the player works as expected, but that is rarely the case. Either 1 or 2 happens, 1 more    frequently than 2. 

- Clicking on timestamp link doesn't move the player to the clicked timestamp

### Comments Section
Adding to features to the comments section would probably be a site-wide change.
The only change specific to Podcasts that I can think of would be the ability to reference timestamps (like in YouTube)

### Additional Features
Aside from what was described above:
- UI redesign. The layout feels awkward, and things just look a bit out-of-place. 
- Related to above, on mobile, the small buttons are awkward to use on the player
- Adding more controls than just "Play" to the big Podcast Logo
- Changing the "loading" text to perhaps a spinning wheel or something else that's visual. Having text there looks out of place
- This is a more long-term thing, but probably having a specific section of dev.to for Podcasts.  This could allow for people to more easily search for specific subject-related podcasts, as well as  recommendations based on what tags they like. 

Obviously, this is not an exhaustive list, and of course, feel free to comment and critique.

**Comment by jessleenyc** [comment]
I agree with all of those thoughts. This would be my priority list:

- Look into and fix why some podcasts aren't playing at all and get rid of the need for the message you see here: https://dev.to/jsjabber/mjs-107-dan-fernandez (in fact, this podcast actually plays so we probably shouldn't show a message here)
- Change podcast relationship so they can belong to users
- Notify user if someone comments on their podcast, give it all the commenting functionality that articles have
- Fix the player issues @vinf100 addressed.

**Comment by benhalpern** [comment]
@lightalloy I decided you'd probably do a better job of writing out the next few issues based on our discussions so I'm going to instead just leave a list of notes here:

- Podcast appearance "tagging", e.g. an association between a user and a podcast episode one could pin to profiles
  - podcast appearance would probably have different states of "displayed", "verified", etc. Ultimately the user should be able to determine if an appearance tag is valid and whether they'd want it displayed on their profile.
- Podcast "ownership" Users should be able to act as admins for podcasts—creating new podcasts, editing info for existing ones etc. Maybe always associate podcasts with organizations and share the role ownership?
- I think the last mile on this whole thing is the player. Let's not assume it's simply about fixing existing bugs because we might want to just simplify our overall approach on the playing in some capacity. Let's touch base once it comes time to make client-side steps.

**Comment by cmgorton** [comment]
@benhalpern is this issue related at all to what the MLH fellows are working on? Working on triaging issues.

**Comment by cmgorton** [comment]
I think this would be a good discussion post on [forem.dev](forem.dev). I know we have had some recent work on the podcast from the MLH fellows. 

Since this one issue has been around for a while without much traction I am going to close it and encourage anyone who would like to see this feature in the future to open a discussion in [forem.dev](forem.dev) . There we can get more eyes on potential features and highlight any we really would love the community to contribute too. Thanks!
